### DISCUSSION TITLE
API missing data after upgrading from 0.5.5 to 0.8.0

### DISCUSSION THREAD

**Discussion by jaya-kumar-softobiz** [ory/kratos#2104] [Q&A] [answered]
Hi, We've been using Ory kratos 0.5.5 version in QA Env and Prod. Everything was working fine after some tweaks.
But recently upgraded QA Ory version to 0.8 and all the identities registered so far are simply lost !!!
Can I do anything now ? 
What is the correct way to upgrade so that data( user identities ) is not lost ? I have to do the upgrade on Prod now but absolutely not want to lose data
Thanks in advance

**Comment by Benehiko** [comment]
Hi @jaya-kumar-softobiz 

Thank you for reaching out and I'm so sorry to hear that! This is not normal behaviour from Kratos and should never be a problem. I will investigate and get back to you as soon as possible. 
/cc @vinckr 

There shouldn't be any destructive actions with doing the migrations. The best thing to do is to always have a backup of your database which you can restore from when this happens. Try in your testing environment to upgrade from v0.5.5 to the next major release, e.g. v0.6.x all the way until v0.8.x.

**Reply by jaya-kumar-softobiz** [reply]
Thank you so much for such a quick reply. Yeah, we lost data in QA, Prod is still fine.
So you mean I should upgrade to 0.5 -> 0.6 -> 0.7 -> 0.8 instead of 0.5 -> 0.8 ?

**Reply by Benehiko** [reply]
yes in the case of such a major problem. Kratos v0.8 should do the migrations, but if for whatever reason it does not, then doing a manual upgrade might show you which version has a problem or introduced this bug.

**Reply by jaya-kumar-softobiz** [reply]
Ok, I will try this stair case upgrade in my local and let you know in case I find anything. Thank you

**Comment by aeneasr** [comment, marked as answer]
Hello, that sounds horrific! Can you confirm that the database tables are really empty? Or are the APIs returning empty lists? We have a lot of very stable tests that ensure that this can not happen! It is important though that you have never made manual changes to the database.

Please also, always when upgrading, read for every version the changelog with immense care: https://github.com/ory/kratos/blob/master/CHANGELOG.md

My hunch says that you might have forgotten to run the kratos migrate sql command.

Unless your database is truly empty, the data should still be safe & sound! Hope this helps :)

**Reply by jaya-kumar-softobiz** [reply]
Oh ok, I did not really run any command for migration, I just updated the version from 0.5.5 to 0.8 and it just pulled the new image. I thought it will handle it automatically.
Do I have to run some commands manually ?
Also I am checking data through APIs. I will ask the dev ops guys to check the DB directly and update you. However, APIs are returning the new registered identities after the upgrade 
Can you please provide some info or links regarding this migration commands please

**Reply by aeneasr** [reply]
Please read the linked changelog and documentation: https://www.ory.sh/kratos/docs/next/cli/kratos-migrate-sql/

May I also recommend potentially checking out Ory Cloud where we run Ory Kratos for you so you can focus on building your app instead of running migrations and figuring out email sending :) You can even get a year for free if you sign up for the start up program an https://www.ory.sh/pricing

**Reply by jaya-kumar-softobiz** [reply]
That sounds really cool. I will inform my manager about this. Its not really my decision :) Thank you for the help.

**Reply by aeneasr** [reply]
You’re welcome :) If this solution worked for you can you mark it as the solution? :)

**Reply by jaya-kumar-softobiz** [reply]
Hey, the old data is still there in the DB... Kratos identities list command is not returning the old data as you guessed !
So I will downgrade to 0.5.5-alpha.1 and then upgrade step by step and let you know. Thank you

**Reply by vinckr** [reply]
Thanks for this @jaya-kumar-softobiz ! 
This would be very valuable to other users as well who might face a similar issue upgrading.

**Reply by aeneasr** [reply]
@vinckr can you maybe write an upgrade guide and add it to the docs?

**Reply by jaya-kumar-softobiz** [reply]
Upgrade guide would be awesome

**Reply by vinckr** [reply]
I added something in this PR https://github.com/ory/kratos/pull/2132
It is not too much yet, what else do you think would be helpful?
@jaya-kumar-softobiz

**Reply by jaya-kumar-softobiz** [reply]
Awesome.

**Reply by jaya-kumar-softobiz** [reply]
@vinckr Obviously, you guys are the pros, looks like you covered everything
